daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/PState.scala

```diff
@@ -108,12 +113,16 @@ object MPState {
 class MPState private () {
 
   val arrayIndexStack = MStackOfLong()
```

Comment:
It is kind of iterating over one array and creating a new array which may or may not be the same length. How about naming these two to be `arraySourcePos` and `arrayTargetPos`?

Reply:
I'm not sure source and target are obvious to me, especially since it could be confusing depending on parse vs unparse. For example, when parsing the source is data but when unparsing the source is the infoset. But along the same lines, maybe using infoset/data would make the distinction more obivous? So maybe something like arrayDataPos and arrayInfosetPos?

@mbeckerle, I skimmed the DFDL spec but couldn't really find any obvious terms to steal from DFDL. Maybe DFDL already has concepts for this kind of thing?

Reply:
I went ahead and put in the first suggestion of `arrayIterationIndexStack` and `arrayIterationPos`. We can easily refactor the name if we come up with a better one.